### PR TITLE
Kth Smallest Element of Binary Tree

### PR DESCRIPTION
**Pull request by Nikitha2309**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given a Binary Tree , Find its k th smallest element

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4319

### CODE REVIEW

File: C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
+/*
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp]: Remove the extra one line space

File: C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+Argument/Return Type
+Input of total no.of nodes is taken
+Input of key values of nodes of tree are taken in level order form 
+Incase of a null node , -1 is taken as input
+Function to find and return kth smallest element
+
+*/
+
+//  Code / Solution
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp]: Remove this redundant comments and one new line

File: C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    }
+    //If no.of nodes of tree is less than k , return -1
+    //else return kth smallest element
+    if(heap.size()<k)
+      return -1;
+    else
+      return heap.top();
+}
+    
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp]: Remove the extra one line space

File: C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+Input Format : 
+Example :
+Enter total no.of nodes of the input Tree ( including NULL nodes ) : 7      
+Enter value of each node of the tree in level order ( if a node is NULL , enter -1 ) with spaces
+12 9 17 8 10 -1 18
+Enter the value of k : 7
+
+Output Format :
+Example : ( Output to the above input example ) 
+Given Tree has less than k nodes for given value of k
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/Binary_Tree_Kth_Smallest_Element.cpp]: Remove the extra extra sample test case one is enough.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Nikitha2309** [comment]
@ankitaggarwal23 @HarshCasper  Requested changes Done . Review and merge it now.
Thank you

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by Nikitha2309** [comment]
@ankitaggarwal23 May I get gssoc tag as well. Just to avoid any issues in evaluation and all.

**Pull request merged**